financial-templates-lib/logger/SlackTransport.js

```diff
@@ -23,86 +44,101 @@ const slackFormatter = info => {
         text: {
           type: "mrkdwn",
           text: ` • _${key}_:\n`
         }
       });
+      // For each key value pair within the object, spread the object out for formatting.
       for (const subKey in info[key]) {
-        // for each key value pair within the object, spread.
-        // Note that only a transaction can come at this level. This is because a transaction is always
-        // an object with the event emmited included in the object.
-        if (subKey == "tx") {
+        // If the length of the value is 66 then we know this is a transaction hash. Format accordingly.
+        if (info[key][subKey].length == 66) {
+          formattedResponse.blocks[
+            formattedResponse.blocks.length - 1
+          ].text.text += `    - _tx_: ${createEtherscanLinkMarkdown(info[key][subKey])}\n`;
+        }
+        // If the length of the value is 42 then we know this is an address. Format accordingly.
+        else if (info[key][subKey].length == 42) {
           formattedResponse.blocks[
             formattedResponse.blocks.length - 1
-          ].text.text += `    - _tx_: <https://etherscan.io/tx/${info[key][subKey]}|${info[key][subKey]}>\n`;
+          ].text.text += `    - _${subKey}_: ${createEtherscanLinkMarkdown(info[key][subKey])}\n`;
         }
         // If the value within the object itself is an object we dont want to spread it any further. Rather,
         // convert the object to a string and print it along side it's key value pair.
-        if (typeof info[key][subKey] === "object" && info[key][subKey] !== null) {
+        else if (typeof info[key][subKey] === "object" && info[key][subKey] !== null) {
           formattedResponse.blocks[
             formattedResponse.blocks.length - 1
           ].text.text += `    - _${subKey}_: ${JSON.stringify(info[key][subKey])}\n`;
+          // Else if not a address, transaction or object then print as ` - key: value`
         } else {
           formattedResponse.blocks[
             formattedResponse.blocks.length - 1
           ].text.text += `    - _${subKey}_: ${info[key][subKey]}\n`;
         }
       }
-    } else {
-      // if the key is a transaction object, generate a transaction link
-      if (key == "tx") {
-        formattedResponse.blocks.push({
-          type: "section",
-          text: {
-            type: "mrkdwn",
-            text: ` • _tx_: <https://etherscan.io/tx/${info[key]}|${info[key]}>\n`
-          }
-        });
-      } else if (key == "mrkdwn") {
-        formattedResponse.blocks.push({
-          type: "section",
-          text: {
-            type: "mrkdwn",
-            text: ` ${info[key]}`
-          }
-        });
+      // Else, if the input is not an object then print the values as key value pairs. First check for addresses or txs
+    } else if (info[key]) {
+      // like with the previous level, if there is a value that is a transaction or an address format accordingly
+      if (info[key].length == 66) {
```

Comment:
nit: should we also check that it starts with `0x` just in case we happen to have a string that's that many characters?

Reply:
Sure.